dotCMS/src/main/java/com/liferay/portal/servlet/ImageServlet.java

```diff
@@ -55,144 +52,49 @@
 	public void init(ServletConfig sc) throws ServletException {
 		synchronized (ImageServlet.class) {
 			super.init(sc);
-
-			ServletContext ctx = getServletContext();
-
-			_companyId = ctx.getInitParameter("company_id");
 		}
 	}
 
 	public void service(HttpServletRequest req, HttpServletResponse res)
-		throws IOException, ServletException {
-
-		String path = req.getPathInfo();
-
-		// The image id may be passed in as image_id, img_id, or i_id
-
-		String imageId = ParamUtil.getString(req, "image_id");
-
-		if (Validator.isNull(imageId)) {
-			imageId = ParamUtil.getString(req, "img_id");
-
-			if (Validator.isNull(imageId)) {
-				imageId = ParamUtil.getString(req, "i_id");
-			}
-		}
-
-		// Company Logo
-
-		if (path.startsWith("/company_logo")) {
-			if (ParamUtil.get(req, "png", false)) {
-				imageId += ".png";
-
-				res.setContentType("image/png");
-			}
-			else if (ParamUtil.get(req, "wbmp", false)) {
-				imageId += ".wbmp";
-
-				res.setContentType("image/vnd.wap.wbmp");
-			}
-		}
-
-		// Image Gallery
-
-		if (path.startsWith("/image_gallery")) {
-			if (!imageId.equals(StringPool.BLANK) &&
-				!imageId.startsWith(_companyId + ".image_gallery.")) {
-
-				imageId = _companyId + ".image_gallery." + imageId;
-
-				if (ParamUtil.get(req, "small", false)) {
-					imageId += ".small";
-				}
-				else {
-					imageId += ".large";
-				}
-			}
-		}
-
-		// Journal Article
-
-		if (path.startsWith("/journal/article")) {
-			if (!imageId.equals(StringPool.BLANK) &&
-				!imageId.startsWith(_companyId + ".journal.article.")) {
-
-				imageId = _companyId + ".journal.article." + imageId;
-
-				String version = req.getParameter("version");
-
-				if (Validator.isNotNull(version)) {
-					imageId += "." + version;
-				}
-			}
-		}
-
-		// Journal Template
-
-		if (path.startsWith("/journal/template")) {
-			if (!imageId.equals(StringPool.BLANK) &&
-				!imageId.startsWith(_companyId + ".journal.template.")) {
-
-				imageId = _companyId + ".journal.template." + imageId;
-				imageId += ".small";
+			throws IOException, ServletException {
+
+		Company company = APILocator.getCompanyAPI().getDefaultCompany();
+
+		//If the City column does not startsWith /dA means that we need to migrate the old logo
+		//to a new dotAsset
+		if (company.getCity() == null || !company.getCity().startsWith("/dA")) {
+			final String imageId = ParamUtil.getString(req,"img_id");
+			final Image image = ImageLocalUtil.get(imageId);
+
+			final File outputFile = new File("company_logo.png");

```

Comment:
This should be written to a tmp file - as it stands it will write it to the root of the dotCMS install

Reply:
This whole process should happen in the init method of the servlet.  Make sure that the servlet is not set to autoload.

Reply:
done